development-vm/replication/sync-aws-elasticsearch.sh

```diff
@@ -0,0 +1,102 @@
+#! /usr/bin/env bash
+
+set -eu
+
+USAGE_LINE="$0 [options] [index_name ...]"
+USAGE_DESCRIPTION="Download Elasticsearch index archives and import into another instance.
+
+If one or more index names are given, only those index files are imported;
+otherwise, all index files are imported."
+. ./common-args.sh
+if $SKIP_ELASTIC; then
+  exit
+fi
+
+shift $(($OPTIND-1))
+
+LOCAL_ES_HOST="http://localhost:9200/"
+LOCAL_ARCHIVE_PATH="${DIR}/elasticsearch"
+
+status "Starting search index replication from AWS"
+
+if $SKIP_DOWNLOAD; then
+  status "Skipping fetch of new archives"
+
+  if [ ! -d $LOCAL_ARCHIVE_PATH ]; then
+    error "Download directory ${LOCAL_ARCHIVE_PATH} does not exist and fetching new archives has been skipped."
+    exit 1
+  fi
+else
+  if [ -d $LOCAL_ARCHIVE_PATH ]; then
+    ok "Download directory ${LOCAL_ARCHIVE_PATH} exists"
+  else
+    status "Creating local backups directory"
+    mkdir -p $LOCAL_ARCHIVE_PATH
+    ok "Download directory ${LOCAL_ARCHIVE_PATH} created"
+  fi
+
+  status "Downloading latest es backups from AWS"
+
+  remote_file_details=$(aws s3 ls s3://govuk-integration-database-backups/elasticsearch/indices/)
+  remote_paths=$(echo $remote_file_details | ruby -e 'STDOUT << STDIN.read.split("PRE").group_by { |n| n.split(/-\d/).first }.map { |_, d| d.sort.last.strip }.join(" ")')
+  for remote_path in $remote_paths; do
+    status "Syncing data from ${remote_path}"
+    mkdir $LOCAL_ARCHIVE_PATH/$remote_path
+    aws s3 sync s3://govuk-integration-database-backups/elasticsearch/indices/${remote_path} $LOCAL_ARCHIVE_PATH/$remote_path/ --only-show-errors
+  done
+
+  exit 0
+fi
+
+exit 1
```

Comment:
Was this used for debugging? It looks like it was left in by accident.

Reply:
This was removed in a later update